test/libsolidity/semanticTests/modifiers/function_return_parameter_complex.sol

```diff
@@ -0,0 +1,26 @@
+// Test to see if the function return parameter, when forwarded to the modifier actually has value
+// zero.
+contract A {
+    modifier alwaysZeros(uint256 a, uint256 b) {
+        _;
+        require(a == 0, "a is not zero");
+        require(b == 0, "b is not zero");
```

Comment:
Are the require statements executed? Doesn't control flow terminate with `_;`?

Reply:
No. You can sort of imagine the modifiers as  replacing `_;` by `f_return_parameters := f_body_only(f_arguments)`. The rest of the statements would continue running. Even when the function has an early `return ;`. See `transitionNext` modifier in https://docs.soliditylang.org/en/v0.8.1/common-patterns.html?highlight=state%20machine#state-machine